spec.bs

```diff
@@ -2078,23 +2078,44 @@ The <dfn http-header><code>Ad-Auction-Allowed</code></dfn> HTTP response header
 [=structured header=] whose value must be a [=structured header/boolean=].
 
 <div algorithm>
-To <dfn>validate fetching response</dfn> given a [=response=] |response|, null, failure, or a
-[=byte sequence=] |responseBody|, and a [=string=] |mimeType|:
+To <dfn>validate fetching response headers</dfn> given a [=response=] |response|:
 
-  1. If |responseBody| is null or failure, return false.
   1. If [=header list/getting a structured field value|getting=] [:Ad-Auction-Allowed:] and
-    "`item`" from |response|'s [=response/header list=] does not return a true value, return false.
+     "`item`" from |response|'s [=response/header list=] does not return a true value, return false.
+  1. If |response|'s [=response/status=] is not an [=ok status=], return false.
+  1. Return true.
+</div>
+
+<div algorithm>
+To <dfn>validate fetching response mime and body</dfn> given a [=response=] |response|, null,
+failure, or a [=byte sequence=] |responseBody|, and a [=string=] |mimeType|:
+
+  1. If |responseBody| is null or failure, return false.
   1. Let |headerMimeType| be the result of [=header list/extracting a MIME type=] from |response|'s
     [=response/header list=].
   1. Return false if any of the following conditions hold:
     * |headerMimeType| is failure;
     * |mimeType| is "`text/javascript`" and |headerMimeType| is not a [=JavaScript MIME type=];
     * |mimeType| is "`application/json`" and |headerMimeType| is not a [=JSON MIME type=].
-  1. Let |mimeTypeCharset| be |headerMimeType|'s [=MIME type/parameters=]["`charset`"].
-  1. Return false if any of the following conditions hold:
-    * |mimeTypeCharset| does not [=map/exist=], or |mimeTypeCharset| is "utf-8", and |responseBody|
-      is not [=UTF-8=] encoded;
-    * |mimeTypeCharset| is "us-ascii", and not all bytes in |responseBody| are [=ASCII bytes=].
+    * |mimeType| is "`application/wasm`" and |headerMimeType|'s [=MIME type/essence=] is not

```

Comment:
our code has [a comment](https://crsrc.org/c/content/services/auction_worklet/public/cpp/auction_downloader.cc;drc=c0265133106c7647e90f9aaa4377d28190b1a6a9;l=109) of not using header mime type, but directly parse from headers. What was the difference of the two? Just trying to figure out if we need to do something similar to [this spec](https://webassembly.github.io/spec/web-api/#compile-a-potential-webassembly-response), or not.

Reply:
Hmm, looks like the code that's calling this calls that method in WASM WebAPI spec.... so the checks weren't actually missing, so maybe I should just remove this portion.

Reply:
... or not. It's a different one of similarly named methods.

Reply:
And done..